[Bug] savepoint time over 1min, then cacel job field

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### Java Version

openjdk11

### Scala Version

2.12.x

### StreamPark Version

v2.0.0

### Flink Version

1.14.2

### deploy mode

kubernetes-session

### What happened

When canceling the job through the streampark ui, select the savepoint, and then throw a timeout exception.

From the streampark ui, you can see that the status of the job is field. After a while, when the real savepoint of the flink job succeeds, the status on the ui changes to canceld again.

This will cause the job to fail to resume from the latest saved savepoint

### Error Exception

```log
java.util.concurrent.CompletionException: java.lang.reflect.InvocationTargetException
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1702)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.reflect.InvocationTargetException
	at jdk.internal.reflect.GeneratedMethodAccessor963.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.streampark.flink.client.FlinkClient$.$anonfun$cancel$1(FlinkClient.scala:73)
	at org.apache.streampark.flink.proxy.FlinkShimsProxy$.$anonfun$proxy$1(FlinkShimsProxy.scala:63)
	at org.apache.streampark.common.util.ClassLoaderUtils$.runAsClassLoader(ClassLoaderUtils.scala:40)
	at org.apache.streampark.flink.proxy.FlinkShimsProxy$.proxy(FlinkShimsProxy.scala:63)
	at org.apache.streampark.flink.client.FlinkClient$.cancel(FlinkClient.scala:68)
	at org.apache.streampark.flink.client.FlinkClient.cancel(FlinkClient.scala)
	at org.apache.streampark.console.core.service.impl.ApplicationServiceImpl.lambda$cancel$6(ApplicationServiceImpl.java:1235)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	... 3 more
Caused by: java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1886)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2021)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.cancelJob(FlinkSubmitTrait.scala:482)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.cancelJob$(FlinkSubmitTrait.scala:451)
	at org.apache.streampark.flink.client.impl.KubernetesNativeSessionSubmit$.org$apache$streampark$flink$client$trait$KubernetesNativeSubmitTrait$$super$cancelJob(KubernetesNativeSessionSubmit.scala:48)
	at org.apache.streampark.flink.client.trait.KubernetesNativeSubmitTrait.doCancel(KubernetesNativeSubmitTrait.scala:99)
	at org.apache.streampark.flink.client.trait.KubernetesNativeSubmitTrait.doCancel$(KubernetesNativeSubmitTrait.scala:82)
	at org.apache.streampark.flink.client.impl.KubernetesNativeSessionSubmit$.doCancel(KubernetesNativeSessionSubmit.scala:121)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.cancel(FlinkSubmitTrait.scala:159)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.cancel$(FlinkSubmitTrait.scala:143)
	at org.apache.streampark.flink.client.impl.KubernetesNativeSessionSubmit$.cancel(KubernetesNativeSessionSubmit.scala:48)
	at org.apache.streampark.flink.client.FlinkClientHandler$.cancel(FlinkClientHandler.scala:46)
	at org.apache.streampark.flink.client.FlinkClientHandler.cancel(FlinkClientHandler.scala)
	... 14 more
```


### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java
+++ b/streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java
@@ -244,6 +244,7 @@ public class Application implements Serializable {
   private transient String savePoint;
   private transient Boolean savePointed = false;
   private transient Boolean drain = false;
+  private transient Long savePointTimeout = 60L;
   private transient Boolean allowNonRestored = false;
   private transient String socketId;
   private transient String projectName;

--- a/streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java
+++ b/streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java
@@ -1244,6 +1244,7 @@ public class ApplicationServiceImpl extends ServiceImpl<ApplicationMapper, Appli
             appParam.getSavePointed(),
             appParam.getDrain(),
             customSavepoint,
+            application.getSavePointTimeout(),
             application.getK8sNamespace());
 
     final Date triggerTime = new Date();

--- a/streampark-console/streampark-console-webapp/src/api/flink/app/app.type.ts
+++ b/streampark-console/streampark-console-webapp/src/api/flink/app/app.type.ts
@@ -147,6 +147,7 @@ interface AppControl {
 export interface CancelParam {
   id: string;
   savePointed: boolean;
+  savePointTimeout: number;
   drain: boolean;
   savePoint: string;
 }

--- a/streampark-console/streampark-console-webapp/src/views/flink/app/components/AppView/StopApplicationModal.vue
+++ b/streampark-console/streampark-console-webapp/src/views/flink/app/components/AppView/StopApplicationModal.vue
@@ -23,6 +23,7 @@
 </script>
 
 <script setup lang="ts" name="StopApplicationModal">
+  import { InputNumber } from 'ant-design-vue';
   import { BasicForm, useForm } from '/@/components/Form';
   import { SvgIcon } from '/@/components/Icon';
   import { BasicModal, useModalInner } from '/@/components/Modal';
@@ -69,6 +70,16 @@
         ifShow: ({ values }) => !!values.stopSavePointed,
       },
       {
+        field: 'savePointTimeout',
+        label: 'SavePoint Timeout',
+        component: 'Select',
+        afterItem: () => h('span', { class: 'conf-switch' }, 'savepoint timeout, Unit: second'),
+        slot: 'savePointTimeout',
+        defaultValue: 60,
+        ifShow: ({ values }) => !!values.stopSavePointed,
+        required: true,
+      },
+      {
         field: 'drain',
         label: 'Drain',
         component: 'Switch',
@@ -90,12 +101,14 @@
   /* submit */
   async function handleSubmit() {
     try {
-      const { stopSavePointed, drain, customSavepoint } = (await validate()) as Recordable;
+      const { stopSavePointed, customSavepoint, savePointTimeout, drain } =
+        (await validate()) as Recordable;
       const stopReq = {
         id: app.id,
         savePointed: stopSavePointed,
-        drain: drain,
         savePoint: customSavepoint,
+        savePointTimeout: savePointTimeout,
+        drain: drain,
       };
 
       if (stopSavePointed) {
@@ -151,9 +164,12 @@
   >
     <template #title>
       <SvgIcon name="shutdown" style="color: red" />
-
       {{ t('flink.app.view.stop') }}
     </template>
-    <BasicForm @register="registerForm" class="!pt-20px" />
+    <BasicForm @register="registerForm" class="!pt-20px">
+      <template #savePointTimeout="{ model, field }">
+        <InputNumber v-model:value="model[field]" :min="10" :max="7200" step="10" />
+      </template>
+    </BasicForm>
   </BasicModal>
 </template>

--- a/streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/bean/CancelRequest.scala
+++ b/streampark-flink/streampark-flink-client/streampark-flink-client-api/src/main/scala/org/apache/streampark/flink/client/bean/CancelRequest.scala
@@ -31,4 +31,5 @@ case class CancelRequest(flinkVersion: FlinkVersion,
                          override val withSavepoint: Boolean,
                          withDrain: Boolean,
                          savepointPath: String,
+                         savePointTimeout: Long,
                          override val kubernetesNamespace: String = K8sFlinkConfig.DEFAULT_KUBERNETES_NAMESPACE) extends SavepointRequestTrait

--- a/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala
+++ b/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala
@@ -475,16 +475,14 @@ trait FlinkClientTrait extends Logger {
 
     val savePointDir: String = tryGetSavepointPathIfNeed(cancelRequest)
 
-    val clientTimeout = getOptionFromDefaultFlinkConfig(cancelRequest.flinkVersion.flinkHome, ClientOptions.CLIENT_TIMEOUT)
-
     val clientWrapper = new FlinkClusterClient(client)
 
     (Try(cancelRequest.withSavepoint).getOrElse(false), Try(cancelRequest.withDrain).getOrElse(false)) match {
       case (false, false) =>
         client.cancel(jobID).get()
         null
-      case (true, false) => clientWrapper.cancelWithSavepoint(jobID, savePointDir).get(clientTimeout.toMillis, TimeUnit.MILLISECONDS)
-      case (_, _) => clientWrapper.stopWithSavepoint(jobID, cancelRequest.withDrain, savePointDir).get(clientTimeout.toMillis, TimeUnit.MILLISECONDS)
+      case (true, false) => clientWrapper.cancelWithSavepoint(jobID, savePointDir).get(cancelRequest.savePointTimeout, TimeUnit.SECONDS)
+      case (_, _) => clientWrapper.stopWithSavepoint(jobID, cancelRequest.withDrain, savePointDir).get(cancelRequest.savePointTimeout, TimeUnit.SECONDS)
     }
   }
 

